File: Fabric/src/main/java/hantonik/atomic/core/block/AtomicDropExperienceBlock.java
package hantonik.atomic.core.block;

import net.minecraft.util.valueproviders.ConstantInt;
import net.minecraft.util.valueproviders.IntProvider;
import net.minecraft.world.level.block.DropExperienceBlock;

import java.util.function.Function;

public class AtomicDropExperienceBlock extends DropExperienceBlock {
    public AtomicDropExperienceBlock(Function<Properties, Properties> properties) {
        this(properties, ConstantInt.of(0));
    }

    public AtomicDropExperienceBlock(Function<Properties, Properties> properties, IntProvider xpRange) {
        super(properties.apply(Properties.of()), xpRange);
    }
}

File: Forge/src/main/kotlin/hantonik/atomic/core/block/AtomicDropExperienceBlock.kt
package hantonik.atomic.core.block

import net.minecraft.util.valueproviders.ConstantInt
import net.minecraft.util.valueproviders.IntProvider
import net.minecraft.world.level.block.DropExperienceBlock

open class AtomicDropExperienceBlock(properties: (Properties) -> Properties, xpRange: IntProvider) : DropExperienceBlock(properties.invoke(Properties.of()), xpRange) {
    constructor(properties: (Properties) -> Properties) : this(properties, ConstantInt.of(0))
}